ts/image-overlay.directive.ts

```diff
@@ -21,33 +21,193 @@ import { TRANSPARENT_PIXEL } from './consts';
 import { LayerGroupProvider } from './layer-group.provider';
 import { LayerProvider } from './layer.provider';
 
+/**
+ * Angular2 directive for Leaflet image overlays.
+ *
+ * *You can use this directive in an Angular2 template after importing `YagaModule`.*
+ *
+ * How to use in a template:
+ * ```html
+ * <yaga-map>
+ *     <yaga-image-overlay
+ *         [(url)]="..."
+ *         [(display)]="..."
+ *         [(opacity)]="..."
+ *         [(zIndex)]="..."
+ *         [(bounds)]="..."
+ *         [(north)]="..."
+ *         [(east)]="..."
+ *         [(south)]="..."
+ *         [(west)]="..."
+ *
+ *         (add)="..."
+ *         (remove)="..."
+ *         (popupopen)="..."
+ *         (popupclose)="..."
+ *         (tooltipopen)="..."
+ *         (tooltipclose)="..."
+ *         (click)="..."
+ *         (dbclick)="..."
+ *         (mousedown)="..."
+ *         (mouseover)="..."
+ *         (mouseout)="..."
+ *         (contextmenu)="..."
+ *         (loading)="..."
```

Comment:
not implemented

+ *         (loading)="..."
+ *         (tileunload)="..."
+ *         (tileloadstart)="..."
+ *         (tileerror)="..."
+ *         (tileload)="..."
+ *         (load)="..."